### PR TITLE
display accurate failure message instead of just erroring

### PR DESCRIPTION
**Pull request by nickpell**
fixes a longstanding bug where the scheduled report fails hard if its owner has been removed

@dannyroberts

### CODE REVIEW

File: corehq/apps/reports/models.py [hunk 1]
```diff
@@ -467,6 +467,13 @@ def get_report_content(self, lang, attach_excel=False):
                 file_obj = dispatch_func(render_as='excel')
             else:
                 file_obj = None
+            if response.status_code == 302:
+                return _("We are sorry, but your saved report '%(config_name)s' "
+                         "is no longer accessible because the owner %(username)s "
+                         "is no longer active.") % {
+                    'config_name': self.name,
+                    'username': self.owner.username
+                }, None
```
Comment (czue) [inline corehq/apps/reports/models.py]: just musing - this double return value "what's the second arg all about" thing would be much nice with a `namedtuple`

File: corehq/apps/reports/models.py [hunk 2]
```diff
@@ -467,6 +467,13 @@ def get_report_content(self, lang, attach_excel=False):
                 file_obj = dispatch_func(render_as='excel')
             else:
                 file_obj = None
+            if response.status_code == 302:
+                return _("We are sorry, but your saved report '%(config_name)s' "
+                         "is no longer accessible because the owner %(username)s "
+                         "is no longer active.") % {
+                    'config_name': self.name,
+                    'username': self.owner.username
```
Comment (czue) [inline corehq/apps/reports/models.py]: are there ever any concerns about `owner` being `None`?

**Comment by czue** [comment]
no blockers, feel free to follow up on comments as you want.

File: corehq/apps/reports/models.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/reports/models.py]: but ... the code is pep8 ... that means it's good code right? :stuck_out_tongue_winking_eye:

File: corehq/apps/reports/models.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/reports/models.py]: That should only be an issue if the ```WebUser``` that created the report is deleted, which doesn't seem likely.  In either case no email should get sent, just right now we'd log an exception.

File: corehq/apps/reports/models.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/reports/models.py]: https://github.com/dimagi/commcare-hq/pull/8239

**Pull request merged**